packages/amplify-cli/src/extensions/amplify-helpers/remove-resource.ts

```diff
@@ -116,14 +115,14 @@ const deleteResourceFiles = async (context: $TSContext, category: string, resour
     const { allResources } = await context.amplify.getResourceStatus();
     allResources.forEach((resourceItem) => {
       if (resourceItem.dependsOn) {
-        resourceItem.dependsOn.forEach((dependsOnItem) => {
+        for (const dependsOnItem of resourceItem.dependsOn) {
```

Comment:
This change is out of scope of this PR, but just found this bug when reading the code....

Reply:
why was this a bug?

Reply:
`forEach` doesn't break if `throw new`